### PR TITLE
Fixed code quality issues

### PR DESCRIPTION
**Pull request by withshubh**
#### Description
Hi :wave: 
I ran the [DeepSource](https://deepsource.io) analysis on the forked copy of this repo and found some interesting [code quality issues](https://deepsource.io/gh/withshubh/DeepCoMP/issues/?category=recommended).

#### Summary of changes

- Removed unnecessary use of comprehension
- Used literal syntax instead of function calls to create data structure
- Changed methods not using its bound instance to staticmethods
- Used `is` to compare type of objects
- Added .deepsource.toml

### CODE REVIEW

File: deepcomp/env/entities/user.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def __init__(self, id, map, pos_x, pos_y, movement, dr_req=1):
         self.movement = movement
         self.dr_req = dr_req
         # dict of connected BS: BS (only connected BS are keys!) --> data rate of connection
-        self.bs_dr = dict()
+        self.bs_dr = {}
```
Comment (stefanbschneider) [inline deepcomp/env/entities/user.py:32]: Why is initializing with `{}` better than with `dict()`? I always feel like `dict()` is more explicit. With `{}`, I sometimes get confused whether it initializes a dict or a set.

File: deepcomp/env/single_ue/base.py [hunk 2]
```diff
@@ -294,7 +296,8 @@ def step_reward(self, rewards):
         ue = self.ue_list[(self.time-1) % self.num_ue]
         return rewards[ue]
 
-    def done(self):
+    @staticmethod
+    def done():
```
Comment (stefanbschneider) [inline deepcomp/env/single_ue/base.py:300]: Not sure if that's a good idea. `done(self)` is overwritten in the child classes. What happens if it's declared as static in the parent class?

**Review by stefanbschneider — COMMENTED** [review]
Hi, thanks for your PR.

Could you elaborate what happens when adding `.deepsource.toml` to the project?

Also did you test the code with your changes to see if it still works? I made a few small comments in some code sections.

Once this is clarified, I'm happy to merge.

File: deepcomp/env/entities/user.py (hunk 1 above)
Comment (withshubh) [reply deepcomp/env/entities/user.py:32]: Fixing this improves the minor performance. Please have a look [here](https://deepsource.io/gh/withshubh/DeepCoMP/issue/PTC-W0019/description)

File: deepcomp/env/single_ue/base.py (hunk 2 above)
Comment (withshubh) [reply deepcomp/env/single_ue/base.py:300]: Okay I'll revert the fix if you want but have a look at [this](https://deepsource.io/gh/withshubh/DeepCoMP/issue/PYL-R0201/description/).

File: .deepsource.toml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+version = 1
```
Comment (stefanbschneider) [inline .deepsource.toml:1]: What does adding this file do? Will it run something (DeepSource?) automatically through some GitHub actions?

File: deepcomp/env/single_ue/base.py (hunk 2 above)
Comment (stefanbschneider) [reply deepcomp/env/single_ue/base.py:300]: Ok thanks. I think the memory impact isn't that relevant here. Could you please revert it in this case?

In the other cases, the fixes to add `@staticmethod` look good to me and can stay. Thanks.

Comment (withshubh) [reply deepcomp/env/single_ue/base.py]: Done! :sparkles:

File: .deepsource.toml (hunk 3 above)
Comment (withshubh) [reply .deepsource.toml:1]: [DeepSource](https://deepsource.io/) works as an additional service that you don’t have to run yourself. All you have to do is just integrate DeepSource in the repo and create a configuration for DeepSource (`.deepsource.toml` file) and leave the rest on DeepSource :sunglasses: 

Here are the steps to do it:
- Install DeepSource on your repository [here](https://deepsource.io/signup).
- Activate analysis [here](https://deepsource.io/gh/CN-UPB/DeepCoMP/).

**Comment by stefanbschneider** [comment]
Thanks, the PR is good to go then! I'll give DeepSource a try :)

**Pull request merged**